### PR TITLE
Feat: add change network feature

### PR DESCRIPTION
**Pull request by braianj**
Screen Shot 2022-08-08 at 11 33 40

### CODE REVIEW

**Review by 2fd — APPROVED** [review]

File: src/components/errors/ErrorNetworkMismatch.tsx [hunk 1]
```diff
@@ -1,33 +1,35 @@
 import React from 'react'
 import { Button } from 'decentraland-ui/dist/components/Button/Button'
 import { ErrorContainer, ErrorDetails, ErrorImage } from './Error'
-import { reload } from "./utils"
-import { NETWORK } from '../../integration/url'
-
 import errorImage from '../../images/errors/robotsmiling.png'
+import { switchToChainId } from '../../eth/provider'
 import './errors.css'
+import { getChainName } from '@dcl/schemas/dist/dapps/chain-id'
```
Comment (nicosantangelo) [inline src/components/errors/ErrorNetworkMismatch.tsx:7]: Maybe we want to move this to the top (below `'decentraland-ui/dist/components/Button/Button'`) to keep consistent with other file's order

File: src/eth/provider.ts [hunk 2]
```diff
@@ -55,3 +56,16 @@ export async function disconnect(): Promise<void> {
     return
   }
 }
+
+export async function switchToChainId(chainId: ChainId) {
+  try {
+    const provider = await connection.getProvider()
+
+    await switchProviderChainId(provider, chainId)
+    window.location.reload()
```
Comment (nicosantangelo) [inline src/eth/provider.ts:65]: I was wondering, is this reload necessary? I mean, if we're changing the current provider, is it possible to just do ir without reloading the entire page?

**Review by nicosantangelo — APPROVED** [review]
LGTM, but comments about two minor things

File: src/components/errors/ErrorNetworkMismatch.tsx [hunk 3]
```diff
@@ -4,12 +4,21 @@ import { getChainName } from '@dcl/schemas/dist/dapps/chain-id'
...
 import errorImage from '../../images/errors/robotsmiling.png'
 import { switchToChainId } from '../../eth/provider'
+import { track } from '../../utils/tracking'
 import './errors.css'
 
 export interface ErrorNetworkMismatchProps {
   extra: Record<string, any>
 }
 
+const switchChain = (options: Record<string, any>) => {
```
Comment (nicosantangelo) [inline src/components/errors/ErrorNetworkMismatch.tsx:14]: Isn't the typing of options a bit loose? We could call this function like this:

```ts
switchChain({ something: 22, breaking: { stuff: 2 } })
```

Typescript'd say everything's fine, but the call would blow up. Isn't it better to just receive `wantedChainId` and `providerChainId`?

Comment (nicosantangelo) [inline src/components/errors/ErrorNetworkMismatch.tsx:14]: Is this function going to be (possibly) needed elsewhere? it seems like it being local to this file might be a problem creating repeated code if any other piece of code needs it

File: src/components/errors/ErrorNetworkMismatch.tsx [hunk 4]
```diff
@@ -4,12 +4,21 @@ import { getChainName } from '@dcl/schemas/dist/dapps/chain-id'
 import { ErrorContainer, ErrorDetails, ErrorImage } from './Error'
 import errorImage from '../../images/errors/robotsmiling.png'
 import { switchToChainId } from '../../eth/provider'
+import { track } from '../../utils/tracking'
 import './errors.css'
 
 export interface ErrorNetworkMismatchProps {
   extra: Record<string, any>
```
Comment (nicosantangelo) [inline src/components/errors/ErrorNetworkMismatch.tsx:11]: The same problem pointed out [in line 14](https://github.com/decentraland/explorer-website/pull/258/commits/61913d62b0486a7064c9a25c1a7fe2442ca5cb8b#diff-1ac65470fc2c1a0d5c5cab3c4d9ee33c59a1c27e700b17bedbd0f59443c96634R14) is present here by the way. This seems unnecessarily loose as a type

**Review by nicosantangelo — COMMENTED** [review]
A few comments!

**Review by nicosantangelo — APPROVED** [review]
Piola!

**Pull request merged**